packages/grid/x-data-grid/src/components/base/GridErrorHandler.tsx

```diff
@@ -38,20 +21,11 @@ function GridErrorHandler(props: { children: React.ReactNode }) {
       logger={logger}
       render={(errorProps) => (
         <GridMainContainer>
-          <GridAutoSizer
-            nonce={rootProps.nonce}
-            disableHeight={rootProps.autoHeight}
-            onResize={handleResize}
-          >
-            {() => (
-              <ErrorOverlayWrapper>
-                <rootProps.components.ErrorOverlay
-                  {...errorProps}
-                  {...rootProps.componentsProps?.errorOverlay}
-                />
-              </ErrorOverlayWrapper>
-            )}
-          </GridAutoSizer>
+          <rootProps.components.ErrorOverlay
+            {...errorProps}
+            {...rootProps.componentsProps?.errorOverlay}
+            rowHeight={rootProps.rowHeight}
```

Comment:
Two things:

- To get the row height prefer `gridDensityRowHeightSelector`. The prop is not aware of the current density.
- You could use `useGridRootProps` inside the overlay, instead of passing the prop.

Reply:
Didn't think about density, thanks!